### PR TITLE
[DOCFIX] revise docs in advanced/

### PR DESCRIPTION
**Pull request by Xenorith**

### CODE REVIEW

File: docs/en/advanced/Scalability-Tuning.md [hunk 1]
```diff
@@ -17,86 +17,97 @@ aims to enumerate a list of such parameters which should be tuned for the scale
...
 
 ### Thread Pool Size
+
 An executor pool is used on the master to handle concurrent client requests. If you expect a large
 number of concurrent clients communicating with the master, tune the thread pool size by modifying
 the following properties. The actual thread count depends on the maximum concurrency expected on the
 cluster. For example, if spawning `10000` concurrent tasks (clients) each with a client thread
 pool size of `4`, set the master thread pool max to greater than `40000`.
+
+TODO: wait aren't these properties for this master <-> worker communication, not master <-> client communication?
```
Comment (Xenorith) [inline docs/en/advanced/Scalability-Tuning.md]: halp

File: docs/en/advanced/Scalability-Tuning.md [hunk 2]
```diff
@@ -17,86 +17,97 @@ aims to enumerate a list of such parameters which should be tuned for the scale
...
-Alluxio master JVM to spawn the number of threads specified in the previous section: `kernel.pid_max`,
-`vm.max_map_count`, max user processes limit (`ulimit -u`) and open files limit (`ulimit -n`).
 
 An exception message like `java.lang.OutOfMemoryError: unable to create new native thread`
-indicates that the limits may need tuning.
+indicates that operating system limits may need tuning.
 
-Note:
+Several parameters limit the number of threads that a process can spawn:
+TODO: how does one configure the first two???
```
Comment (Xenorith) [inline docs/en/advanced/Scalability-Tuning.md]: halp

File: docs/en/advanced/Scalability-Tuning.md [hunk 3]
```diff
@@ -17,86 +17,97 @@ aims to enumerate a list of such parameters which should be tuned for the scale
...
+- `kernel.pid_max`
+- `vm.max_map_count`
+- Max user process limit: `ulimit -u`
+- Max open files limit: `ulimit -n`
 
-* As a rule of thumb, `vm.max_map_count` should be at least twice the limit for master threads.
+These limits are often set for the particular user that launch the Alluxio process.
+As a rule of thumb, `vm.max_map_count` should be at least twice the limit for master threads.
+TODO: twice the limit for master threads, so what determines the limit for master threads???
```
Comment (Xenorith) [inline docs/en/advanced/Scalability-Tuning.md]: halp

File: docs/en/advanced/Scalability-Tuning.md [hunk 4]
```diff
@@ -17,86 +17,97 @@ aims to enumerate a list of such parameters which should be tuned for the scale
...
 ## RPC Retry Interval
 
-If frequent timeouts are observed communicating with the Alluxio master, tune the rpc retry intervals:
+The following properties tune RPC retry intervals:
+
 ```properties
 alluxio.user.rpc.retry.max.duration=2min
 alluxio.user.rpc.retry.base.sleep.ms=1s
 ```
+TODO: `alluxio.user.rpc.retry.base.sleep.ms` is awkwardly named if it takes a duration and not a number of ms as the value? is this correct?
```
Comment (Xenorith) [inline docs/en/advanced/Scalability-Tuning.md]: double check please

File: docs/en/advanced/Security.md [hunk 5]
```diff
@@ -11,13 +11,15 @@ priority: 0
 
 This document describes the following security related features in Alluxio.
 
-1. [Authentication](#authentication): If `alluxio.security.authentication.type=SIMPLE` (by default),
+1. [Authentication](#authentication): 
+TODO: I can't even begin to parse this...
```
Comment (Xenorith) [inline docs/en/advanced/Security.md]: i may know how to parse this now, after reading the rest of the doc...

File: docs/en/advanced/Security.md [hunk 6]
```diff
@@ -102,21 +106,25 @@ drwxr-xr-x jack           staff                       24       PERSISTED 11-20-2
...
+This implementation executes the `groups` shell command to fetch the group memberships of the given user.
+The user group mapping is cached, with an expiration period configured by the
+`alluxio.security.group.mapping.cache.timeout` property, with a default value of `60s`.
+If set to a value of `0`, caching is disabled.
 
-Property `alluxio.security.authorization.permission.supergroup` defines a super group. Any users
-belong to this group are also super users.
+The `alluxio.security.authorization.permission.supergroup` property defines a super group.
+Any users belong to this group are also super users.
+TODO: but what is a super user?
```
Comment (Xenorith) [inline docs/en/advanced/Security.md]: we never define super user anywhere :/

File: docs/en/advanced/Security.md [hunk 7]
```diff
@@ -102,21 +106,25 @@ drwxr-xr-x jack           staff                       24       PERSISTED 11-20-2
...
 
 ### Initialized directory and file permissions
 
-The initial creation permission is 777, and the difference between directory and file is 111.
-For default umask value 022, the created directory has permission 755 and file has permission 644.
-The umask can be set by property `alluxio.security.authorization.permission.umask`.
+When a file is created, it is initally assigned fully opened permissions of `666` by default.
+Similarly, a directory is initally assigned with `777` permissions. TODO: i think that's what it previously meant by difference of 111 but someone please check??? i also think it was mixed up according to the following example
```
Comment (Xenorith) [inline docs/en/advanced/Security.md]: double check please

File: docs/en/advanced/Security.md [hunk 8]
```diff
@@ -128,131 +136,145 @@ The owner, group, and permissions can be changed by two ways:
...
+
+After enabling impersonation on the master for a given user,
+the client must indicate which user it wants to impersonate.
+This is configured by the `alluxio.security.login.impersonation.username` property.
+
+If the property is set to an empty string or `_NONE_`, impersonation is disabled.
+If the property is set to `_HDFS_USER_`, the Alluxio client will impersonate as the same user as
+the HDFS client when using the Hadoop compatible client.
+
+TODO: just wondering, can we set this to a particular string indicating the specific user name? if possible, i find it weird that it's not explained here...
```
Comment (Xenorith) [inline docs/en/advanced/Security.md]: halp

File: docs/en/advanced/Tiered-Locality.md [hunk 9]
```diff
@@ -47,81 +53,82 @@ To configure tiered locality, follow these steps:
...
 
-If the user does nothing to provide tiered identity info, each entity will
-perform a localhost lookup to set its node-level identity info. If other locality tiers
-are left unset, they will not be used to inform locality decisions. To set
-the value for a locality tier, set the configuration property
+If the user does nothing to provide tiered identity info,
+each entity will perform a hostname lookup to set its node-level identity info.
+If other locality tiers are left unset, they will not be used to inform locality decisions.
+To set the value for a locality tier, set the configuration property:
+TODO: huh why is this empty???
```
Comment (Xenorith) [inline docs/en/advanced/Tiered-Locality.md]: needs filling in

File: docs/en/advanced/Performance-Tuning.md [hunk 10]
```diff
@@ -166,71 +170,67 @@ consider reducing this value so that checkpoints happen more often.
...
 
-`alluxio.worker.network.netty.rpc.threads.max` - Default value is 2048, consider increasing this value
-if you have a very large number of concurrent client accesses and are seeing threadpool exhausted
-errors on the worker.
+The number of RPC threads available on a worker is configured by
+`alluxio.worker.network.netty.rpc.threads.max`, with a default value of `2048`.
+This value should be increased if exhausted thread pool errors are being logged on workers.
 
 ### Async eviction and watermarks
+TODO: this is the exact same topic as "Evicting Stale Data" in Alluxio Storage Management. only keep one???
```
Comment (Xenorith) [inline docs/en/advanced/Performance-Tuning.md]: ??

File: docs/en/advanced/Performance-Tuning.md [hunk 11]
```diff
@@ -262,20 +262,18 @@ will be triggered frequently when space utilization gets close to the high water
...
-into resource contention on the master.
+On a single client, the number of threads connecting to the master is configured by the
+`alluxio.user.block.master.client.threads` and `alluxio.user.file.master.client.threads` properties,
+each with a default value of 10.
+The size of the master thread pool that serves connections to clients should be tuned to match
+the maximum number of concurrrent client connections.
+For example, if the master expects up to 100 clients, each with the default number of connections,
+the master's thread pool should be configured to be at least 100 * 10 * 2 = 2000.
+ 
+TODO: what is the property key for configuring the master thread pool size??? And shouldn't this be doubled since there are two sets of threads for blocks and files?
```
Comment (Xenorith) [inline docs/en/advanced/Performance-Tuning.md]: halp?

File: docs/en/advanced/Performance-Tuning.md [hunk 12]
```diff
@@ -262,20 +262,18 @@ will be triggered frequently when space utilization gets close to the high water
...
 
 ### Loading UFS Metadata
+TODO: Wasn't this just explained in the above section called "UFS Path Cache"??? duplicate sections >.<
```
Comment (Xenorith) [inline docs/en/advanced/Performance-Tuning.md]: ??

File: docs/en/advanced/Performance-Tuning.md [hunk 13]
```diff
@@ -307,6 +305,7 @@ Compared with the UFS sync feature, loading metadata only discovers new files fr
 not modify any updated metadata of files, such as updated timestamps, or deleted files.
 
 ### Syncing UFS Metadata
+TODO: this too is explained previously in UFS Metadata Sync under Namespace Management. cmon...
```
Comment (Xenorith) [inline docs/en/advanced/Performance-Tuning.md]: ??

File: docs/en/advanced/Scalability-Tuning.md [hunk 14]
```diff
@@ -17,86 +17,97 @@ aims to enumerate a list of such parameters which should be tuned for the scale
...
+Increase the interval to reduce the number of heartbeat checks.
 
 ## Alluxio Worker Configuration
 
 ### Heartbeat Intervals and Timeouts
 
-Frequent heartbeats can cause delays. Tune the following parameter(s) to control the frequency with
-which all workers heartbeat with the master.
+The frequency in which a worker checks in with the master is set by the following properties:
+TODO: I feel like there should be an explanation on why there's two separate heartbeat intervals...
```
Comment (Xenorith) [inline docs/en/advanced/Scalability-Tuning.md]: yes/no?

File: docs/en/advanced/Scalability-Tuning.md (hunk 1 above)
Comment (apc999) [reply docs/en/advanced/Scalability-Tuning.md]: the naming of properties is bad. it is actually for master<->client communitcation

File: docs/en/advanced/Scalability-Tuning.md (hunk 4 above)
Comment (apc999) [reply docs/en/advanced/Scalability-Tuning.md]: removed. no `ms` needed here

File: docs/en/advanced/Scalability-Tuning.md (hunk 14 above)
Comment (apc999) [reply docs/en/advanced/Scalability-Tuning.md]: yes. added comments

File: docs/en/advanced/Security.md (hunk 5 above)
Comment (apc999) [reply docs/en/advanced/Security.md]: rephrased a bit

File: docs/en/advanced/Security.md (hunk 6 above)
Comment (apc999) [reply docs/en/advanced/Security.md]: a def added

File: docs/en/advanced/Security.md (hunk 7 above)
Comment (apc999) [reply docs/en/advanced/Security.md]: correct

File: docs/en/advanced/Security.md (hunk 8 above)
Comment (apc999) [reply docs/en/advanced/Security.md]: added

File: docs/en/advanced/Performance-Tuning.md (hunk 10 above)
Comment (apc999) [reply docs/en/advanced/Performance-Tuning.md]: this is actually the default behavior now. I will remove this session

**Pull request merged**